src/platform/lumin-runtime/platform-factory.js

```diff
@@ -42,6 +57,31 @@ export class PlatformFactory extends NativeFactory {
     return element instanceof MxsPrismController;
   }
 
+  _setCallbackDataPerNode (node, callbackId, eventName) {
+    const prismId = executor.callNativeFunction(node, 'getPrismId');
+    if (this._eventCallbackData[prismId] === undefined) {
+      this._eventCallbackData[prismId] = {};
+    }
+    const nodeId = executor.callNativeFunction(node, 'getNodeId');
+    if (this._eventCallbackData[prismId][nodeId] === undefined) {
+      this._eventCallbackData[prismId][nodeId] = {};
+    }
+
+    this._eventCallbackData[prismId][nodeId][eventName] = callbackId;
+  }
+
+  _getCallbackDataPerNode (node) {
+    const prismData = this._eventCallbackData[executor.callNativeFunction(node, 'getPrismId')];
```

Comment:
If you replace this with line with a call to this._getCallbackDataPerPrism there's a bit less duplication and you don't need the conditional below (since it falls back on an empty object).

Reply:
You're right, thank you @tonycuadra !